File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        if len(s)==0 or (len(s) ==0 and len(t)==0):
            return True

        if len(t) == 0:
            return False

        p1 = 0
        p2 =0

        count = 0
        for p2 in range(len(t)):
            #move pointer p1 if the letters are equal
            if t[p2] == s[p1]:
                count +=1
                p1 +=1
            
            if count == len(s):
                return True

        return False  

        
        

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {

        //Base Case
        int l1 = s.length();
        int l2 = t.length();
        
        if(l1 ==0 || (l1 ==0 && l2==0)){
            return true;
        }

        if(l2 ==0){
            return false;
        }

        char[] tt = t.toCharArray();
        char[] ss = s.toCharArray();
        int p1 = 0;
        int count = 0;

        for(int p2 =0;p2<l2;p2++){

            if(tt[p2] == ss[p1]){
                count++;
                p1++;
            }

            if(p1 == l1){
                return true;
            }
        }

        return false;
        
    }
}